crates/runtime/src/instance.rs

```diff
@@ -781,13 +773,12 @@ impl InstanceHandle {
     /// the `wasmtime` crate API rather than this type since that is vetted for
     /// safety.
     pub unsafe fn new(
-        module: Arc<Module>,
+        module: Arc<dyn Deref<Target = Module>>,
```

Comment:
This makes me a little uncomfortable using `Deref` like this. 

The only reason for this, I believe, is keeping the `CodeMemory` alive, right? Since the memory management of `InstanceHandle` is detached from the handle itself (it happens in `Store` for the `wasmtime` crate), could the managment of `CodeMemory` move into `Store` as well?

For example could each `CompiledModule` store an `Arc<CodeMemory>`, and then we push a strong reference of that into a `Store` whenever a module is instantiated within a `Store`?

Reply:
I replaced `Deref<Target = Module>` with `InstanceContext`. I'm not convinced that `Store` is good location for `CodeMemory`. I understand that the end goal is to be capable to ditch entire `Module`/`CodeModule` even if `Instance` or its exports are present, but keep `CodeMemory` alive. Can we do it in follow up PRs?